### PR TITLE
Payload deletion support in storage layer

### PR DESCRIPTION
**Pull request by conor10**
This PR adds support for deleting entries in the Constellation payload store.
This is the first PR to address the following [request](https://github.com/jpmorganchase/constellation/issues/22) 

The API changes to support this method, and receive/sendRaw and will follow.

### CODE REVIEW

File: Constellation/Node/Storage/BerkeleyDb.hs [hunk 1]
```diff
@@ -118,6 +120,14 @@ load db mtx k = do
         Nothing -> Left "Key not found in BerkeleyDb payload.db"
         Just v  -> Right $ decode $ BL.fromStrict v
 
+delete :: Db
+     -> Maybe DbTxn
+     -> Text
+     -> IO ()
+delete db mtx k = do
```
Comment (patrickmn) [inline Constellation/Node/Storage/BerkeleyDb.hs]: `delete db mtx k = void $ db_del [] db mtx (TE.encodeUtf8 k)`

File: Constellation/Node/Storage/Memory.hs [hunk 2]
```diff
@@ -44,6 +46,10 @@ load mvar k = atomically $ do
         Nothing -> Left "Key not found in memory database"
         Just v  -> Right v
 
+delete :: Db -> Text -> IO ()
+delete mvar k = atomically $ do
```
Comment (patrickmn) [inline Constellation/Node/Storage/Memory.hs]: `delete mvar k = atomically $ void $ modifyTVar mvar (HM.delete k)`

File: Constellation/Node/Storage/Sqlite.hs [hunk 3]
```diff
@@ -75,6 +76,13 @@ load c k = do
         []       -> Left "load: No payload found"
         _        -> Left "load: More than one payload found"
 
+delete :: Connection -> Text -> IO ()
+delete c k = do
```
Comment (patrickmn) [inline Constellation/Node/Storage/Sqlite.hs]: maybe use void and remove do

File: test/Constellation/Node/Storage/BerkeleyDb/Test.hs [hunk 4]
```diff
@@ -1,8 +1,23 @@
 {-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE Strict #-}
```
Comment (patrickmn) [inline test/Constellation/Node/Storage/BerkeleyDb/Test.hs]: Switched over to StrictData in other files at least until the difference can be benchmarked (Strict may be too aggressive, force evaluation of log arguments even if log statements are muted, etc.)

File: test/Constellation/Node/Storage/TestUtil.hs [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE Strict #-}
+module Constellation.Node.Storage.TestUtil where
+
+import ClassyPrelude
+import Test.Tasty.HUnit ((@?), (@?=), Assertion)
+import qualified Data.ByteString.Char8 as C
```
Comment (patrickmn) [inline test/Constellation/Node/Storage/TestUtil.hs]: Please call this BC

**Comment by conor10** [comment]
Coolio - done.

**Pull request merged**